### PR TITLE
UI, obs-ffmpeg: Set audio track titles for FFmpeg Output

### LINKED ISSUE #7658: audio track name missing when recording ffmpeg
### Operating System Info

Windows 10

### Other OS

_No response_

### OBS Studio Version

28.0.3

### OBS Studio Version (Other)

_No response_

### OBS Studio Log URL

https://link.storjshare.io/jw75lt32aa7axz6kafkdtxahhmdq/obslog%2F2022-10-25%2004-50-01.txt

### OBS Studio Crash Log URL

_No response_

### Expected Behavior

The name of the audio tracks when recording ffmpeg must correspond to the names specified in the audio section

### Current Behavior

The name of the audio tracks when recording ffmpeg does not correspond to the name specified in the audio section

### Steps to Reproduce

-

### Anything else we should know?

_No response_

### PR DESCRIPTION
**Pull request by tuduweb**
### Description

1. Set audio tracks name to `settings` when setup FFmpeg output
2. Add `const char *audio_stream_names[MAX_AUDIO_MIXES]` member to `struct ffmpeg_cfg`.
3. Set *tracks name* from step1 `settings` to (`struct ffmpeg_cfg`) `config`.
4. Set *tracks name* in `config` to FFmpeg *stream metadata* `title` when valid(not null) via `av_dict_set`.

### Motivation and Context

fixes #7658 
same as Standard Recording
set audio stream title via `av_dict_set` *title* in `ffmpeg-mux` via command-line
https://github.com/obsproject/obs-studio/blob/c648222332d1a7ce7ea55fd14fbe514a5f258f33/plugins/obs-ffmpeg/ffmpeg-mux/ffmpeg-mux.c#L538

### How Has This Been Tested?

Win10, latest OBS
via VLC

- audio mux 1,2,3,4

**SETTINGS** audio track names[1-4]:
mux, desk, mic, [empty]
before:
before-Screenshot 2023-02-19 155636

after:
after-Screenshot 2023-02-19 155142

- audio mux 1,4,5

**SETTINGS** audio track names[1-5]:
mux, desk, mic, track4, track5

145-Screenshot 2023-02-19 160249

### Types of changes

 - Bug fix (non-breaking change which fixes an issue) 
 - New feature (non-breaking change which adds functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Review by RytoEX — CHANGES_REQUESTED** [review]

File: UI/window-basic-main-outputs.cpp [hunk 1]
```diff
@@ -1596,8 +1596,27 @@ inline void AdvancedOutput::SetupFFmpeg()
 		config_get_int(main->Config(), "AdvOut", "FFAEncoderId");
 	const char *aEncCustom =
 		config_get_string(main->Config(), "AdvOut", "FFACustom");
+
+	obs_data_array_t *audio_names = obs_data_array_create();
```
[the remark this answers is not in the document]
Comment (tuduweb) [reply UI/window-basic-main-outputs.cpp]: Thx. I have updated this PR.

(I noticed this last week but didn't find this implementation at the time)

**Review by RytoEX — APPROVED** [review]
Seems fine. Does what it set out to do.

File: UI/window-basic-main-outputs.cpp [hunk 2]
```diff
@@ -1984,8 +1984,25 @@ inline void AdvancedOutput::SetupFFmpeg()
 		config_get_int(main->Config(), "AdvOut", "FFAEncoderId");
 	const char *aEncCustom =
 		config_get_string(main->Config(), "AdvOut", "FFACustom");
+
+	OBSDataArrayAutoRelease audio_names = obs_data_array_create();
+
+	for (size_t i = 0; i < MAX_AUDIO_MIXES; i++) {
+		string cfg_name = "Track";
```
Comment (PatTheMav) [inline UI/window-basic-main-outputs.cpp:1991]: As the identifier is used for the configuration file only, there's no need to make this human readable.

`audioTrackName<NUMBER>` would be easier to compose and can start at `0` (though admittedly my alternative is _also_ more human readable).

Comment (RytoEX) [reply UI/window-basic-main-outputs.cpp:1991]: Track<N>Name is how the values are stored in the profile config:
```
[AdvOut]
Track1Name=Main
Track2Name=Mic
Track3Name=Desktop
Track4Name=SomeName4
Track5Name=SomeName5
Track6Name=SomeName6
```

The string built here is used just below to fetch the value from the profile config file.

Comment (PatTheMav) [reply UI/window-basic-main-outputs.cpp:1991]: So that's on us then, too bad. 🙈

Comment (RytoEX) [reply UI/window-basic-main-outputs.cpp:1991]: Indeed. We may wish to change the name at some point in the future, but that's out of scope for this PR.

**Pull request merged**